### PR TITLE
[Improvement](statistics, multi catalog)Implement hive table statistic connector.

### PR DESCRIPTION
**Pull request by Jibing-Li**
This pr is to add the collecting hive statistic function. While the CBO fetching hive table statistics, statistic cache will first load from internal stats olap table. If not found, then using this pr's function to fetch from remote Hive metastore.

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -4041,9 +4041,9 @@ show_param ::=
         RESULT = new ShowTableStatsStmt(tbl, partitionNames);
     :}
     /* show column stats */
-    | KW_COLUMN KW_STATS table_name:tbl opt_col_list:cols opt_partition_names:partitionNames
+    | KW_COLUMN KW_STATS table_name:tbl opt_col_list:cols opt_partition_names:partitionNames opt_properties:properties
     {:
-        RESULT = new ShowColumnStatsStmt(tbl, cols, partitionNames);
+        RESULT = new ShowColumnStatsStmt(tbl, cols, partitionNames, properties);
```
Comment (morningman) [inline fe/fe-core/src/main/cup/sql_parser.cup:4046]: Add document for this new parameter

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.75 seconds
 stream load tsv:          456 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          69.7 seconds inserted 10000000 Rows, about 143K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230626081334_clickbench_pr_167748.html

**Comment by morningman** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 40.46 seconds
 stream load tsv:          458 seconds loaded 74807831229 Bytes, about 155 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          69.7 seconds inserted 10000000 Rows, about 143K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230627021221_clickbench_pr_168246.html

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 41.24 seconds
 stream load tsv:          464 seconds loaded 74807831229 Bytes, about 153 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          69.5 seconds inserted 10000000 Rows, about 143K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230628100553_clickbench_pr_169261.html

**Review by morningman — APPROVED** [review]
LGTM

**Review by Kikyou1997 — APPROVED** [review]

**Pull request merged**